### PR TITLE
Support links encoded in some GeoKrety QRCode

### PR DESCRIPTION
**Pull request by kumy**
@samueltardieu could you please check this when you have time? I used Rx but I'm unsure if I use it right. Thanks

### CODE REVIEW

**Comment by kumy** [comment]
PR updated
Is it too late to put that in `release`?

**Comment by samueltardieu** [comment]
I won't have the time to reread it carefully enough for it to be put in `release` as far as I am concerned.

**Comment by Lineflyer** [comment]
`master` please as we are now in a close to release status.

**Comment by kumy** [comment]
@samueltardieu do have time to reread it?
@Bananeweizen, @rsudev Anyone else would like to review?

**Comment by samueltardieu** [comment]
I didn't plan to reread it. Is it needed?

File: main/src/cgeo/geocaching/connector/trackable/GeokretyConnector.java [hunk 1]
```diff
@@ -191,6 +217,41 @@ public String getTrackableCodeFromUrl(@NonNull final String url) {
...
+    /**
+     * Lookup Trackable Geocode from Tracking Code.
+     *
+     * @param trackingCode
+     *          the Trackable Tracking Code to lookup
+     * @return
+     *          the Trackable Geocode
+     */
+    @Nullable
+    public static String getGeocodeFromTrackingCode(final String trackingCode) {
```
Comment (samueltardieu) [inline main/src/cgeo/geocaching/connector/trackable/GeokretyConnector.java]: It is not called from the UI thread, is it?

Comment (kumy) [inline main/src/cgeo/geocaching/connector/trackable/GeokretyConnector.java]: IIRC original call comes from https://github.com/kumy/cgeo/blob/issue-5135/main/src/cgeo/geocaching/TrackableActivity.java#L213 which itself is called from `onCreate()`

Comment (samueltardieu) [inline main/src/cgeo/geocaching/connector/trackable/GeokretyConnector.java]: If this is the case, this is **not** OK. @Bananeweizen already introduced such a network call on the UI thread recently, and this should not enter our code base.

**Comment by kumy** [comment]
I'm curious on how @Bananeweizen fixed that. @samueltardieu do you remember the class/function where this was implememted?

**Comment by samueltardieu** [comment]
Nope, I don't remember. I just remember that it is done wrong.

**Comment by kumy** [comment]
Found the other code where this was also implemented: https://github.com/cgeo/cgeo/commit/766d09124dbbddb96a991b0db841500416a6dd1e#commitcomment-13330678

File: main/src/cgeo/geocaching/connector/trackable/GeokretyConnector.java (hunk 1 above)
Comment (kumy) [inline main/src/cgeo/geocaching/connector/trackable/GeokretyConnector.java]: @samueltardieu I've looked deeper into call stack of this function. Concerned `getGeocodeFromTrackingCode()` function and calling function `searchTrackable()` are running on network thread. 
Here is the call stack from the caller function. Do you think it's ok? Thanks
capture du 2015-10-22 22 59 56

**Comment by Bananeweizen** [comment]
Can one of the admins verify this patch?

**Comment by kumy** [comment]
(I have to rebase the code on current master).
I'm not sure the network call is on the UI thread, and if it is really, I didn't found a good/clean way to do it. Any help will be appreciated ;)

**Comment by samueltardieu** [comment]
I don't understand why you don't handle that as you handle loading a cache from the network and display it. You can either display minimal data and refresh it as soon as the right information is available, or you can use a "loading" activity which waits for the data to be available.

**Comment by Lineflyer** [comment]
As this PR can no longer be merged I close it.
@kumy If you think different please feel free to reopen, else submit a new PR is still needed.

**Comment by kumy** [comment]
@Lineflyer I'll push a new PR as soon as I'll have time :)

**Pull request closed without merging**